.github/workflows/build-apk.yml

```diff
@@ -48,13 +49,43 @@ jobs:
       - name: Grant execute permission for gradlew
         run: chmod +x ./gradlew
 
-      # Build the APK
+      # Decode the Keystore from secret
+      - name: Decode Keystore
+        env:
+          ANDROID_KEYSTORE_BASE64: ${{ secrets.ANDROID_KEYSTORE_BASE64 }}
+        run: |
+          mkdir -p ~/.android
+          echo "$ANDROID_KEYSTORE_BASE64" | base64 --decode > ~/.android/debug.keystore
+
+        # Generate SHA1 fingerprint from keystore
+      - name: Generate SHA1
+        run: keytool -list -v -keystore ~/.android/debug.keystore -storepass ${{ secrets.KEYSTORE_PASSWORD }} -keypass ${{ secrets.KEYSTORE_PASSWORD }}
+      # Build the debug APK
       - name: Build APK
         run: ./gradlew assembleDebug
 
+      # Find the latest version of build-tools
+      - name: Find latest build-tools version
+        id: build_tools
+        run: echo "build_tools_version=$(ls ${ANDROID_HOME}/build-tools | sort -V | tail -n 1)" >> $GITHUB_ENV
+
+      # Sign the debug APK using apksigner
```

Comment:
An eventual [zipalign](https://developer.android.com/tools/zipalign) could be added to improve memory efficiency, to be done before signing as specified on [apksigner](https://developer.android.com/tools/apksigner)